### PR TITLE
fix(migrations): quarantine workspace_files orphans instead of deleting them

### LINKED ISSUE #357: Workspace FK migration silently deletes orphan rows
`openrag/scripts/migrations/alembic/versions/f1a2b3c4d5e6_add_workspace_files_file_id_fk.py` runs `DELETE FROM workspace_files WHERE file_id NOT IN (SELECT file_id FROM files)` at line 39 and a second `DELETE WHERE file_fk IS NULL` at line 59 inside `upgrade()`. The matching `downgrade()` at lines 86 to 103 only restores the column structure, not the deleted rows, so the migration is irreversibly lossy with no operator opt-in, no row-count log, and no quarantine. If the orphans were caused by an upstream bug, the data is gone before anyone can audit them.

Before each `DELETE`, query `COUNT(*)` and emit it via the alembic logger so operators can review the loss. Better, copy the targeted rows into a `workspace_files_orphans_<rev>` table inside the same transaction; the structure stays small, downgrade can restore from it, and a follow-up cleanup migration can drop the table once operators confirm.

### PR DESCRIPTION
**Pull request by EnjoyBacon7**
Fixes #357

Reopens #906 against `develop`, which is the right base per CONTRIBUTING.md.
That one was closed for base-branch reasons, not for content. Same branch, plus
one review fix.

The workspace_files FK migration deleted rows it could not resolve. It never said
how many, and downgrade could not bring them back.

Now each delete copies the rows into a `workspace_files_orphans_f1a2b3c4d5e6`
table in the same statement, logs the count, and `downgrade()` puts them back.
Also swapped `NOT IN` for `NOT EXISTS`, and the quarantine table is created
behind a `table_exists` guard as CLAUDE.md requires.

Checked with an upgrade/downgrade round trip on a real Postgres, plus 5 new unit tests.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved data safety when workspace file records reference missing or unresolved files.
  - Affected records are preserved for recovery instead of being permanently deleted.
  - Migrations now stop safely when required file references are missing.
  - Repeated migrations avoid duplicating work.
  - Rollbacks restore eligible preserved records while safely handling conflicts.
  - Rollbacks continue gracefully when preserved-record data is unavailable.
  - Preserved records remain available when complete restoration is not possible, with a warning logged.

### CODE REVIEW

File: tests/unit/services/persistence/test_workspace_files_fk_migration.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    monkeypatch.setattr(migration, "op", fake_op)
+    monkeypatch.setattr(migration, "column_type_is", lambda *_: False)
+    for helper in ("column_exists", "index_exists", "fk_exists", "unique_constraint_exists", "table_exists"):
+        monkeypatch.setattr(migration, helper, helpers.get(helper, lambda *_: True))
+    func()
+    return "\n".join(fake_op.calls)
+
+
+def test_upgrade_quarantines_instead_of_deleting(monkeypatch, migration) -> None:
+    statements = _run(monkeypatch, migration, migration.upgrade)
```
[the remark this answers is not in the document]
Comment (EnjoyBacon7) [reply tests/unit/services/persistence/test_workspace_files_fk_migration.py]: Already fixed in 6458c4a3, which landed after the commit this review ran against (543bf8bc). The change is the same one you propose: `table_exists=lambda *_: False` and asserting on the recorded `create_table` call.

Also added `test_upgrade_skips_the_quarantine_table_when_it_already_exists` for the other side of the guard, since idempotency is the point of the CLAUDE.md rule this fix is following. 6 tests pass, and all checks are green.

**Comment by hedhoud** [comment]
Confirmed this edge case. `NOT EXISTS` selects legacy `workspace_files` rows with a NULL `file_id`, but the quarantine table currently rejects them, so the migration can abort before the FK conversion completes. The quarantine path should preserve NULL IDs (or explicitly handle them without violating its constraint) and add regression coverage for a NULL legacy value.

**Review by hedhoud — APPROVED** [review]
Reviewed the latest changes. The migration preserves orphaned rows, rejects invalid NULL file IDs before mutation, retains incomplete downgrade quarantine data, and excludes quarantine tables from autogeneration. All PR checks pass.
